src/v2/services/implementations/MetamaskWalletService.ts

```diff
@@ -126,18 +126,23 @@ export class MetamaskWalletService extends WalletService implements IWalletServi
         })) as string;
         const { uncompressedPubKey } = utils.recoverPublicKeyFromSig(h160Address, msg, signature);
 
-        const contract = new web3.eth.Contract(
-          lockdropDispatchAbi as AbiItem[],
-          evmPrecompiledContract.lockdropDispatch
-        );
+        // Todo: remove this logic once contract address for Shiden lockdrop has been updated to align with Astar and Shibuya
+        const chainId = await web3.eth.getChainId();
+        const isShiden = chainId === 336;
+
+        const contractAddress = isShiden
+          ? '0x0000000000000000000000000000000000005005'
+          : evmPrecompiledContract.lockdropDispatch;
```

Comment:
if it need to be updated at every upgrades, I think we can create objects for each contract address mapped with chainId like this. 

```
evmPrecompiledContract.lockdropDispatch = {
336: '0x0000000000000000000000000000000000005005',
592: '0x0000000000000000000000000000000000005007',
81: '0x0000000000000000000000000000000000005007'
}
```

it's not necessary for now, just a comment.

Reply:
No, the contract address for Shiden (...5005) will be changed to 5007 soon, so we don't need to tweak the UI :D

Reply:
Cool!